src/transformers/models/deit/modeling_deit.py

```diff
@@ -511,6 +516,11 @@ def forward(
         # and head_mask is converted to shape [num_hidden_layers x batch x num_heads x seq_length x seq_length]
         head_mask = self.get_head_mask(head_mask, self.config.num_hidden_layers)
 
+        # TODO: have a better solution
+        expected_dtype = self.embeddings.patch_embeddings.projection.weight.dtype
```

Comment:
Can we have a safety checker from the `ImageProcessor` side instead? We could for example add an argument in `xxxFeatureExtractor` to more generally allow half-precision input

Reply:
But I think `ImageProcessor` is going to be deprecated in the future, so maybe let's leave it like this for now? cc @sgugger

Reply:
It's the feature extractors that will be deprecated ;-)
I think having this check inside the model is not bad, though I'm surprised `autocast` doesn't do it for us?

Reply:
Ohh I see thanks a lot!
I did not tried with autocast, I only tried to load and run the model using (..., torch_dtype=torch.float16), I think it's ok to make it work like that since I suspect autocast will not work with 8-bit models. So maybe let's leave it like this and I'll add a slow test testing inference in fp16 runs fine